e2e/api/steps.js

```diff
@@ -506,16 +503,23 @@ module.exports = {
     await assertError('Hearing', claimantResponseData.invalid.Hearing.moreThanYear,
       'The date cannot be in the past and must not be more than a year in the future');
 
-    await assertSubmittedEvent(expectedCcdState || 'PROCEEDS_IN_HERITAGE_SYSTEM', {
+    await assertSubmittedEvent('AWAITING_APPLICANT_INTENTION', {
       header: 'You have chosen to proceed with the claim',
       body: '>We will review the case and contact you to tell you what to do next.'
     });
+    await waitForFinishedBusinessProcess(caseId);
 
+    //Check camunda has brought it to Judicial Referral
+    const caseForDisplay = await apiRequest.fetchCaseForDisplay(user, caseId);
+    assert.equal(caseForDisplay.state.id, expectedCcdState);
     await waitForFinishedBusinessProcess(caseId);
-    if (!expectedCcdState) {
+
+    if (expectedCcdState == 'PROCEEDS_IN_HERITAGE_SYSTEM') {
       await assertCorrectEventsAreAvailableToUser(config.applicantSolicitorUser, 'PROCEEDS_IN_HERITAGE_SYSTEM');
       await assertCorrectEventsAreAvailableToUser(config.defendantSolicitorUser, 'PROCEEDS_IN_HERITAGE_SYSTEM');
       await assertCorrectEventsAreAvailableToUser(config.adminUser, 'PROCEEDS_IN_HERITAGE_SYSTEM');
+    } else if (expectedCcdState == 'JUDICIAL_REFERRAL') {
+      //await assertCorrectEventsAreAvailableToUser(config.judgeUser, 'JUDICIAL_REFERRAL');
```

Comment:
what if expectedCcdState is neither?

Reply:
Then I am not sure what state it is going to, and what users need to be checked.
I could add "else { await assertCorrectEventsAreAvailableToUser(config.applicantSolicitorUser, expectedCcdState) }"
But what if that user does not actually engage with that state?

Reply:
Taken into account that expectedCcdState can be neither, whilst also continuing to take into account the delay for Judicial Referal state in showing up, by creating delayedCcdState parameter.